src/main/java/com/dmdirc/ui/messages/EventFormatter.java

```diff
@@ -155,4 +155,15 @@ private String applyDisplayProperties(final DisplayPropertyMap displayProperties
         return res.toString();
     }
 
+    /**
+     * Get the displayable properties map for this event according to the formatter.
+     *
+     * @param event Event to get properties for.
+     * @return DisplayPropertyMap from formatter.
+     */
+    public DisplayPropertyMap getFormatDisplayableProperties(final DisplayableEvent event) {
+        return formatProvider.getFormat(event.getClass()).isPresent()
+                ? formatProvider.getFormat(event.getClass()).get().getDisplayProperties()
```

Comment:
Also I don't understand what this is doing.... If it has formats defined in the provider then it uses those and ignores what the event says, otherwise it uses whatever the event has? That doesn't make much sense to me.